src/main/java/com/zebrunner/carina/webdriver/DriverHelper.java

```diff
@@ -496,6 +506,192 @@ public void addCookie(Cookie cookie) {
         });
     }
 
+    /**
+     * add a cookie object into the driver
+     *
+     * @param name name of the cookie
+     * @param value the value of the cookie
+     * @param domain the domain of the cookie
+     * @param path the path of the cookie
+     * @param expiry when the cookie expires
+     */
+    public void addCookie(String name, String value, String domain, String path, Date expiry) {
+        addCookie(new Cookie(name, value, domain, path, expiry));
+    }
+
+    /**
+     * add a cookie object into the driver
+     *
+     * @param name name of the cookie
+     * @param value the value of the cookie
+     * @param domain the domain of the cookie
+     * @param path the path of the cookie
+     * @param expiry when the cookie expires
+     * @param isSecure if the cookie is secure or not
+     * @param isHttpOnly if it is on http or https
+     */
+    public void addCookie(String name, String value, String domain, String path, Date expiry, boolean isSecure, boolean isHttpOnly) {
+        addCookie(new Cookie(name, value, domain, path, expiry, isSecure, isHttpOnly));
+    }
+
+    /**
+     * gets a specific cookie
+     *
+     * @param name the cookie being targeted
+     * @return cookie with the given name
+     * @see WebDriver.Options#getCookieNamed(String)
+     */
+    public Cookie getCookieNamed(String name) {
+        return getDriver().manage().getCookieNamed(name);
+    }
+
+    /**
+     * get the value of a cookie that was a URL-encoded Map.
+     *
+     * @param name the cookie being targeted
+     * @return decoded cookie value
+     */
+    @SuppressWarnings("unchecked")
+    public Map<String, String> getDecodedValueOfCookieNamed(String name) {
+        String cookieJsonString = null;
+        try {
+            cookieJsonString = URLDecoder.decode(getCookieNamed(name).getValue(), StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            LOGGER.error("Error caught while decoding the cookie with the name: " + name, e);
+        }
+        return new Gson().fromJson(cookieJsonString, HashMap.class);
+    }
+
+    public Set<Cookie> getAllCookies() {
+        return getDriver().manage().getCookies();
+    }
+
+    /**
+     * Return a cookie String similar to document.cookie in the browser console
+     * Includes the cookie name and value, and doesn't include the path, domain, etc for each cookie
+     * @return String with all the cookie names and their values
+     */
+    public String getAllCookiesString() {
```

Comment:
Why we need to return cookies as string?

Reply:
Removing this method.